File: src/main/java/com/mock_json/mock_api/models/Project.java
package com.mock_json.mock_api.models;

import jakarta.persistence.*;
import lombok.*;

import org.hibernate.annotations.SoftDelete;
import org.hibernate.annotations.UpdateTimestamp;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonProperty;

import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "projects")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@SoftDelete
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String slug;

    @CreationTimestamp
    @JsonProperty("created_at")
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    @JsonProperty("channel_id")
    private String ChannelId;

    @Column(nullable = true,columnDefinition = "TEXT")
    private String description;

    @UpdateTimestamp
    @JsonProperty("updated_at")
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "team_id")
    @JsonBackReference
    private Team team;


    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "project_id")
    @JsonBackReference
    private ForwardProxy forwardProxy;

    @Column(name = "is_forward_proxy_active", columnDefinition = "BOOLEAN DEFAULT FALSE")
    @JsonProperty("is_forward_proxy_active")
    @Builder.Default
    private Boolean isForwardProxyActive = false;
}

File: internal/models/project.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Project represents a project in the system.
type Project struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
	Name      string         `gorm:"uniqueIndex:idx_project_name_team_id;not null" json:"name"`
	Slug      string         `gorm:"uniqueIndex:idx_project_slug_team_id;not null" json:"slug"`
	TeamID    uint           `gorm:"uniqueIndex:idx_project_name_team_id;uniqueIndex:idx_project_slug_team_id;not null" json:"team_id"`
	Team      *Team          `gorm:"foreignKey:TeamID" json:"team,omitempty"`
}

// TableName specifies the table name for the Project model.
func (Project) TableName() string {
	return "projects"
}
